--- origsrc/mariadb-10.0.24/include/my_global.h	2016-02-17 14:59:54.000000000 -0600
+++ src/mariadb-10.0.24/include/my_global.h	2016-04-19 10:42:20.171362100 -0500
@@ -46,8 +46,6 @@
 #undef __WIN__
 #undef __WIN32__
 #define HAVE_ERRNO_AS_DEFINE
-#define _POSIX_MONOTONIC_CLOCK
-#define _POSIX_THREAD_CPUTIME
 #endif /* __CYGWIN__ */
 
 #if defined(__OpenBSD__) && (OpenBSD >= 200411)
@@ -201,7 +199,7 @@
 #define unlikely(x)	__builtin_expect(((x) != 0),0)
 
 /* Fix problem with S_ISLNK() on Linux */
-#if defined(TARGET_OS_LINUX) || defined(__GLIBC__)
+#if defined(TARGET_OS_LINUX) || defined(__GLIBC__) || defined(__CYGWIN__)
 #undef  _GNU_SOURCE
 #define _GNU_SOURCE 1
 #endif
--- origsrc/mariadb-10.0.24/storage/connect/filamfix.cpp	2016-02-17 14:59:55.000000000 -0600
+++ src/mariadb-10.0.24/storage/connect/filamfix.cpp	2016-03-03 13:10:51.019425900 -0600
@@ -34,6 +34,12 @@
 #endif  // !UNIX
 #include <sys/stat.h>
 #include <fcntl.h>
+#ifdef __CYGWIN__
+#define ftruncate64 ftruncate
+#define lseek64 lseek
+#define open64 open
+#define O_LARGEFILE 0
+#endif
 #endif  // !__WIN__
 
 /***********************************************************************/
--- origsrc/mariadb-10.0.24/storage/connect/filamvct.cpp	2016-02-17 14:59:55.000000000 -0600
+++ src/mariadb-10.0.24/storage/connect/filamvct.cpp	2016-03-03 13:10:51.026426800 -0600
@@ -40,6 +40,12 @@
 #include <io.h>
 #endif  // !UNIX
 #include <fcntl.h>
+#ifdef __CYGWIN__
+#define ftruncate64 ftruncate
+#define lseek64 lseek
+#define open64 open
+#define O_LARGEFILE 0
+#endif
 #endif  // !__WIN__
 
 /***********************************************************************/
--- origsrc/mariadb-10.0.24/storage/connect/reldef.cpp	2016-02-17 14:59:55.000000000 -0600
+++ src/mariadb-10.0.24/storage/connect/reldef.cpp	2016-03-03 13:10:51.654006500 -0600
@@ -514,9 +514,9 @@ PTABDEF OEMDEF::GetXdef(PGLOBAL g)
     } // endif getdef
 #else   // !__WIN__
   const char *error = NULL;
-  Dl_info dl_info;
     
 #if 0  // Don't know what all this stuff does
+  Dl_info dl_info;
   // The OEM lib must retrieve exported CONNECT variables
   if (dladdr(&connect_hton, &dl_info)) {
     if (dlopen(dl_info.dli_fname, RTLD_NOLOAD | RTLD_NOW | RTLD_GLOBAL) == 0) {
--- origsrc/mariadb-10.0.24/storage/connect/xindex.cpp	2016-02-17 14:59:55.000000000 -0600
+++ src/mariadb-10.0.24/storage/connect/xindex.cpp	2016-03-03 13:10:51.663507700 -0600
@@ -25,6 +25,13 @@
 #include <io.h>
 #endif  // !UNIX
 #include <fcntl.h>
+#ifdef __CYGWIN__
+#define ftruncate64 ftruncate
+#define lseek64 lseek
+#define open64 open
+#define off64_t off_t
+#define O_LARGEFILE 0
+#endif
 #endif  // !__WIN__
 
 /***********************************************************************/
--- origsrc/mariadb-10.1.14/storage/mroonga/vendor/groonga/config.h.cmake	2016-05-08 05:49:08.000000000 -0500
+++ src/mariadb-10.1.14/storage/mroonga/vendor/groonga/config.h.cmake	2016-07-05 15:07:35.381448900 -0500
@@ -101,6 +101,7 @@
 #cmakedefine HAVE_LINUX_FUTEX_H
 #cmakedefine HAVE_MEMORY_H
 #cmakedefine HAVE_NETDB_H
+#cmakedefine HAVE_POLL_H
 #cmakedefine HAVE_PTHREAD_H
 #cmakedefine HAVE_SIGNAL_H
 #cmakedefine HAVE_STDARG_H
@@ -109,6 +110,7 @@
 #cmakedefine HAVE_STRING_H
 #cmakedefine HAVE_SYS_MMAN_H
 #cmakedefine HAVE_SYS_PARAM_H
+#cmakedefine HAVE_SYS_POLL_H
 #cmakedefine HAVE_SYS_RESOURCE_H
 #cmakedefine HAVE_SYS_SELECT_H
 #cmakedefine HAVE_SYS_SOCKET_H
