--- origsrc/mariadb-10.1.30/storage/mroonga/vendor/groonga/CMakeLists.txt	2017-12-21 09:48:52.000000000 -0600
+++ src/mariadb-10.1.30/storage/mroonga/vendor/groonga/CMakeLists.txt	2018-02-05 22:56:59.481413100 -0600
@@ -361,10 +361,7 @@ if(NOT ${GRN_WITH_LZ4} STREQUAL "no")
       pkg_check_modules(LIBLZ4 liblz4)
     endif()
     if(LIBLZ4_FOUND)
-      find_library(LZ4_LIBS
-	NAMES ${LIBLZ4_LIBRARIES}
-	PATHS ${LIBLZ4_LIBRARY_DIRS}
-	NO_DEFAULT_PATH)
+      set(LZ4_LIBS ${LIBLZ4_LIBRARIES})
       set(GRN_WITH_LZ4 TRUE)
     else()
       if(${GRN_WITH_LZ4} STREQUAL "yes")
