--- origsrc/mysql-5.0.51/dbug/Makefile.am	2007-11-15 08:06:50.000000000 -0600
+++ src/mysql-5.0.51/dbug/Makefile.am	2008-03-24 20:25:14.500000000 -0500
@@ -38,10 +38,10 @@
 all:            user.t user.ps
 
 user.t:         user.r $(NROFF_INC)
-		-nroff -mm user.r > $@
+		-nroff -mm $(srcdir)/user.r > $@
 
 user.ps:	user.r $(NROFF_INC)
-		-groff -mm user.r > $@
+		-groff -mm $(srcdir)/user.r > $@
 
 output1.r:      factorial
 		./factorial 1 2 3 4 5 | cat > $@
