--- origsrc/mysql-5.0.51/mysql-test/mysql-test-run.pl	2007-11-15 08:06:16.000000000 -0600
+++ src/mysql-5.0.51/mysql-test/mysql-test-run.pl	2008-03-24 20:25:14.515625000 -0500
@@ -70,8 +70,7 @@
 $| = 1; # Automatically flush STDOUT
 
 our $glob_win32_perl=  ($^O eq "MSWin32"); # ActiveState Win32 Perl
-our $glob_cygwin_perl= ($^O eq "cygwin");  # Cygwin Perl
-our $glob_win32=       ($glob_win32_perl or $glob_cygwin_perl);
+our $glob_win32=       ($glob_win32_perl);
 our $glob_netware=     ($^O eq "NetWare"); # NetWare
 
 require "lib/mtr_cases.pl";
@@ -647,12 +646,6 @@
 
   # Find the absolute path to the test directory
   $glob_mysql_test_dir=  cwd();
-  if ( $glob_cygwin_perl )
-  {
-    # Windows programs like 'mysqld' needs Windows paths
-    $glob_mysql_test_dir= `cygpath -m "$glob_mysql_test_dir"`;
-    chomp($glob_mysql_test_dir);
-  }
 
   # In most cases, the base directory we find everything relative to,
   # is the parent directory of the "mysql-test" directory. For source
