--- origsrc/mysql-5.1.32/client/Makefile.am	2009-02-13 18:51:56.000000000 -0600
+++ src/mysql-5.1.32/client/Makefile.am	2009-04-02 00:51:41.050375000 -0500
@@ -73,26 +73,26 @@ mysqldump_SOURCES=              mysqldum
 mysqlimport_SOURCES=		mysqlimport.c
 
 mysqlimport_LDADD =		$(CXXLDFLAGS) $(CLIENT_THREAD_LIBS) \
+				$(top_builddir)/mysys/libmysys.a \
 				@CLIENT_EXTRA_LDFLAGS@ \
-				$(LIBMYSQLCLIENT_LA) \
-				$(top_builddir)/mysys/libmysys.a
+				$(LIBMYSQLCLIENT_LA)
 
 mysqlshow_SOURCES=		mysqlshow.c
 
 mysqlslap_SOURCES=		mysqlslap.c
 mysqlslap_CFLAGS=		-DTHREAD -UUNDEF_THREADS_HACK
 mysqlslap_LDADD =		$(CXXLDFLAGS) $(CLIENT_THREAD_LIBS) \
+				$(top_builddir)/mysys/libmysys.a \
 				@CLIENT_EXTRA_LDFLAGS@ \
-				$(LIBMYSQLCLIENT_LA) \
-				$(top_builddir)/mysys/libmysys.a
+				$(LIBMYSQLCLIENT_LA)
 
 mysqltest_SOURCES=		mysqltest.cc
 mysqltest_CXXFLAGS=		-DTHREAD -UUNDEF_THREADS_HACK
 mysqltest_LDADD =		$(CXXLDFLAGS) $(CLIENT_THREAD_LIBS) \
 				@CLIENT_EXTRA_LDFLAGS@ \
-				$(LIBMYSQLCLIENT_LA) \
 				$(top_builddir)/mysys/libmysys.a \
 				$(top_builddir)/regex/libregex.a \
+				$(LIBMYSQLCLIENT_LA) \
 				$(CLIENT_THREAD_LIBS)
 
 mysql_upgrade_SOURCES=          mysql_upgrade.c \
--- origsrc/mysql-5.1.32/libmysql/Makefile.shared	2009-02-13 18:52:21.000000000 -0600
+++ src/mysql-5.1.32/libmysql/Makefile.shared	2009-03-18 12:56:34.779375000 -0500
@@ -77,7 +77,7 @@ mysysobjects2 =		my_lib.lo mf_qsort.lo
 mysysobjects =		$(mysysobjects1) $(mysysobjects2)
 target_libadd =		$(mysysobjects) $(mystringsobjects) $(dbugobjects) \
  $(sql_cmn_objects) $(vio_objects) $(sqlobjects)
-target_ldflags = -version-info @SHARED_LIB_VERSION@ @LD_VERSION_SCRIPT@ 
+target_ldflags = -version-info @SHARED_LIB_VERSION@ @LD_VERSION_SCRIPT@ -no-undefined
 vio_objects= vio.lo viosocket.lo viossl.lo viosslfactories.lo
 
 BUILT_SOURCES		= link_sources
